src/forms/formHelpers.js

```diff
@@ -347,89 +347,160 @@ const IntervalColumnHeadings = function ({ type }) {
       valid:      true,
-      value:      value,
-      focusedVal: value,
+      focusedVal: this.props.value,
     };
-
-    // Don't need updating
-    this.generic  = generic;
-    this.name     = name;
-    this.className= time + ' ' + type + ' cashflow-column ' + interval;
-    this.interval = interval;
-    this.store    = store;
   }  // End constructor()
   
   handleFocus = ( evnt, inputProps ) => {
     var newState = {
       focused:    true,
-      focusedVal: toMoneyStr( this.state.value )
+      // `value` here is probably a long decimal
+      focusedVal: this.props.format( this.props.value )
     }
-    this.setState(function ( prevState ) { return newState; });
+    this.setState( newState );
   }
   
   handleBlur = ( evnt ) => {
-    this.setState(function ( prevState ) { return { focused: false, valid: true }; });
+    this.setState({ focused: false, valid: true });
   }
 
   handleChange = ( evnt, inputProps ) => {
     var value = inputProps.value,
-        valid = isPositiveNumber( value );
+        valid = this.props.validate( value );
 
     if ( valid ) {
-      var monthly = toMonthlyAmount[ this.interval ]( evnt, value ),
-          obj     = { name: this.generic , value: monthly };
-      this.store( evnt, obj );
+      this.props.store( evnt, value, this.props.interval );
     }
 
-    this.setState(function ( prevState ) { return {...prevState, valid: valid, focusedVal: value}; });
+    this.setState({ valid: valid, focusedVal: value });
   }  // End handleChange()
 
-  componentWillReceiveProps ({ value }) {
-    this.setState(function ( prevState ) { return { value: value }; });
-  }
-
   render() {
 
-    var { value, valid, focused, focusedVal } = this.state;
+    var { valid, focused, focusedVal }  = this.state;
+    var { value, name, className }      = this.props;
 
-    if ( !focused ) { value = toMoneyStr( value ) }
+    // Format correctly when neighbors are updated, if needed
+    if ( !focused ) { value = this.props.format( value ) }
     else { value = focusedVal; }
 
     /** @todo Different class for something 'future' that has a current value that isn't 0 */
     return (
       <Form.Input
         error     = { !valid }
         value     = { value }
-        name      = { this.name }
-        className = { this.className }
-        style     = {{ width: '7em', display: 'inline-block' }}
+        name      = { name }
+        className = { className }
         onChange  = { this.handleChange }
         onFocus   = { this.handleFocus }
         onBlur    = { this.handleBlur }
         type      = { 'number' } />
     );
 
-  }
+  }  // End render()
+
+};  // End ManagedInput
+
+
+
+// /** @todo description
+// *
+// * @todo Make this more generic and make the caller
+// * handle more of the specifics.
+// *
+// * @function
+// * @param {object} props
+// * @property {object} props.__ - explanation
+// *
+// * @returns Component
+// */
+class CashFlowInput extends Component {
```

Comment:
If this isn't being used anymore, I think it should be removed. If you want to reference it, you can always reach back in git's history.

Reply:
Yeah, leaving that was definitely not intentional.